src/components/application_manager/rpc_plugins/vehicle_info_plugin/include/vehicle_info_plugin/vehicle_info_pending_resumption_handler.h

```diff
@@ -32,65 +32,116 @@
+  void TriggerPendingResumption();
+
+  struct PendingSubscriptionsResumption {
+    PendingSubscriptionsResumption(
+        const uint32_t app_id,
+        const uint32_t corr_id,
+        const std::set<std::string>& requested_vehicle_data)
+        : app_id_(app_id)
+        , fake_corr_id_(corr_id)
+        , waiting_for_hmi_response_(false)
+        , requested_vehicle_data_(requested_vehicle_data) {}
+
+    PendingSubscriptionsResumption(const PendingSubscriptionsResumption& copy) =
+        default;
+
+    /**
+     * @brief IsSuccessfullyDone check if SDL restored all required data or not
+     * @return
+     */
+    bool IsSuccessfullyDone() const;
+
+    bool DataWasRequested(const std::string& vd) const;
+    std::set<std::string> NotSubscribedData() const;
+
+    void FillSubscriptionResults(const smart_objects::SmartObject& response);
+    void FillSubscriptionResults();
+
+    void FillRestoredData(
+        const std::set<std::string>& successful_subscriptions);
+
+    uint32_t app_id_;
+    uint32_t fake_corr_id_;
+    bool waiting_for_hmi_response_;
+    std::set<std::string> requested_vehicle_data_;
+    std::set<std::string> restored_vehicle_data_;
+    std::map<std::string, smart_objects::SmartObject> subscription_results_;
   };
 
-  typedef std::pair<VehicleInfoAppExtension, resumption::Subscriber>
-      FreezedResumption;
-  std::queue<ResumptionAwaitingHandling> freezed_resumptions_;
-  std::map<uint32_t, smart_objects::SmartObject> pending_requests_;
+  void SendHMIRequestForNotSubscribed(const PendingSubscriptionsResumption& pending_resumption);
+  void RaiseFinishedPendingResumption(const PendingSubscriptionsResumption& pending_resumption);
+  /**
+   * @brief SubscribeToFakeRequest will create fake subscription for subscriber
+   * ( resumption_data_processor)
+   * Fake request is required only for subscriber subscription.
+   * This request will not be sen't to HMI so it named as fake request.
+   * Fake request contains all data that need to be resumed for the application
+   * When HMI will resopond for any VehicleDara request, PendingSubscriptionsResumption
+   * will go throw all pending resumptions and fill it will received subscriptions.
+   * If certain pending resumption will take all requested subscriptions PendingSubscriptionsResumption
+   * will take this fake request correaltion id and create fake response based on it.
+   * Within fake response it will notify subscriber about resumption status.
+   *
+   * @param app_id applicaiton to pass into subscriber
+   * @param subscriptions list of requested subscriptions
+   * @param subscriber function to subscribe caller for finished resumption
+   * @return pending_resumption with information about fake correlation id for
+   * subscriber, list of vehicle data to subscribe
+   */
+  PendingSubscriptionsResumption SubscribeToFakeRequest(
+      const uint32_t app,
+      const std::set<std::string>& subscriptions,
```

Comment:
@LuxoftAKutsan I believe set of strings is a good candidate for a typedef as you are using it in several places in this class

Reply:
Added typedef to VehicleDataList